### PR TITLE
fix(PoolTogether): tokenFaucets[].apr could be null

### LINKED ISSUE #3855: [Crash] TypeError: Cannot read property 'toFixed' of null
I posted a tweet included pooltogether's link, then Mask report an error.
Here is my tweet: https://twitter.com/WEN6666661/status/1423144265228587008
and i was in latest develop.

> Cannot read property 'toFixed' of null

Error stack:

<pre>TypeError: Cannot read property 'toFixed' of null
    at PoolView (chrome-extension://momgpnhedkjmcmlmmphclokcpiikhbjd/js/chunk.8937.js:1600:75)
    at Ch (chrome-extension://momgpnhedkjmcmlmmphclokcpiikhbjd/js/npm.react-dom.js:232:137)
    at si (chrome-extension://momgpnhedkjmcmlmmphclokcpiikhbjd/js/npm.react-dom.js:259:255)
    at gk (chrome-extension://momgpnhedkjmcmlmmphclokcpiikhbjd/js/npm.react-dom.js:356:346)
    at fk (chrome-extension://momgpnhedkjmcmlmmphclokcpiikhbjd/js/npm.react-dom.js:343:22)
    at ek (chrome-extension://momgpnhedkjmcmlmmphclokcpiikhbjd/js/npm.react-dom.js:342:369)
    at Vj (chrome-extension://momgpnhedkjmcmlmmphclokcpiikhbjd/js/npm.react-dom.js:342:229)
    at Pj (chrome-extension://momgpnhedkjmcmlmmphclokcpiikhbjd/js/npm.react-dom.js:331:476)
    at J (chrome-extension://momgpnhedkjmcmlmmphclokcpiikhbjd/js/1528.js:19979:203)
    at MessagePort.R (chrome-extension://momgpnhedkjmcmlmmphclokcpiikhbjd/js/1528.js:19980:123)</pre>

## Build info
- Version: 1.35.0
- NODE_ENV: production
- target: chromium
- build: stable
- architecture: web
- firefoxVariant: false
- resolution: desktop
- BUILD_DATE: 2021-08-04T13:49:50.523Z
- VERSION: v1.29.12-712-gdd9a3b596

## Git (*):

dd9a3b596 (develop) on tag "v1.35.0"
git@github.com:DimensionDev/Maskbook.git

### PR DESCRIPTION
**Pull request by UncleBill**
closes #3855

### CODE REVIEW

File: packages/maskbook/src/plugins/PoolTogether/UI/PoolView.tsx [hunk 1]
```diff
@@ -277,7 +277,7 @@ export function PoolView(props: PoolProps) {
                                     classes={{ icon: classes.poolIcon }}
                                 />
                                 {t('plugin_pooltogether_apr', {
-                                    apr: pool.tokenFaucets[0].apr.toFixed(2),
+                                    apr: pool.tokenFaucets[0].apr?.toFixed(2) ?? 0,
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/PoolTogether/UI/PoolView.tsx:280]: What if tokenFaucets[0] is undefined?

Comment (UncleBill) [reply packages/maskbook/src/plugins/PoolTogether/UI/PoolView.tsx:280]: There is a check of `pool.tokenFaucets.length !== 0` above, and we can assume that members of`pool.tokenFaucets` aren't nullable.

But, checking the first member of `tokenFaucets` would make the logic more reliable.

**Review by Jack-Works — APPROVED** [review]

File: packages/maskbook/src/plugins/PoolTogether/UI/PoolView.tsx (hunk 1 above)
Comment (UncleBill) [reply packages/maskbook/src/plugins/PoolTogether/UI/PoolView.tsx]: Just updated the code.

File: packages/maskbook/src/plugins/PoolTogether/UI/PoolView.tsx [hunk 2]
```diff
@@ -228,6 +228,8 @@ export function PoolView(props: PoolProps) {
             </Typography>
         )
     }
+    const tokenFaucet = pool.tokenFaucets[0]
+    const tokenFaucetDripToken = pool.tokens.tokenFaucetDripTokens?.[0]
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/PoolTogether/UI/PoolView.tsx:232]:
```suggestion
    const tokenFaucet = first(pool.tokenFaucets)
    const tokenFaucetDripToken = first(pool.tokens.tokenFaucetDripTokens ?? [])
```

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**